internal/services/resource/template_deployment_resource_test.go

```diff
@@ -208,7 +208,11 @@ func (r TemplateDeploymentResource) Destroy(ctx context.Context, clients *client
 func (TemplateDeploymentResource) basicSingle(data acceptance.TestData) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
-  features {}
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
```

Comment:
this indicates there is likely a bug in the resource, why can the resource group not be cleaned up? is there a resource that isn't correctly waiting upon delete?

Reply:
Hi @katbyte, thank you for reviewing.
According to this [Doc](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/template_deployment), `azurerm_template_deployment` will not clean nested resources. Currently, the testcase error is `the Resource Group still contains Resources`. This error is by design with `azurerm_template_deployment` resource. To resolve this error, we have to set `prevent_deletion_if_contains_resources = false`

Reply:
👍